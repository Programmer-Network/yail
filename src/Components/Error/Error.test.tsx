import { fireEvent, render, screen } from "@testing-library/react";
import { vi } from "vitest";

import { Error } from "./";

describe("Error component", () => {
  const errorText = "404 error";
  const titleText = "Page not found.";
  const descriptionText = "The page you are looking for does not exist.";

  test("renders correctly - snapshot test", () => {
    const { asFragment } = render(
      <Error error={errorText} title={titleText} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("renders error, title, and description", () => {
    render(
      <Error
        error={errorText}
        title={titleText}
        description={descriptionText}
      />
    );
    expect(screen.getByText(errorText)).toBeInTheDocument();
    expect(screen.getByText(titleText)).toBeInTheDocument();
    expect(screen.getByText(descriptionText)).toBeInTheDocument();
  });

  test("conditionally renders icon and backToText", () => {
    const backToText = "Go Back";
    render(
      <Error
        error={errorText}
        title={titleText}
        IconProps={{
          iconName: "IconQuestionMark",
          dataTestId: "icon-question-mark"
        }}
        backToText={backToText}
      />
    );
    expect(screen.getByTestId("icon-question-mark")).toBeInTheDocument();
    expect(screen.getByText(backToText)).toBeInTheDocument();
  });

  test("handles click events", () => {
    const onBackToClick = vi.fn();
    const onIconClick = vi.fn();
    render(
      <Error
        error={errorText}
        title={titleText}
        IconProps={{
          iconName: "IconQuestionMark",
          dataTestId: "icon-question-mark"
        }}
        onIconClick={onIconClick}
        backToText='Go Back'
        onBackToClick={onBackToClick}
      />
    );

    fireEvent.click(screen.getByTestId("icon-question-mark"));
    expect(onIconClick).toHaveBeenCalledTimes(1);

    fireEvent.click(screen.getByText("Go Back"));
    expect(onBackToClick).toHaveBeenCalledTimes(1);
  });

  test("applies custom class name", () => {
    const customClass = "custom-error-class";

    render(
      <Error error={errorText} title={titleText} className={customClass} />
    );

    const errorElement = screen.getByTestId("error-wrapper");
    expect(errorElement).toHaveClass(customClass);
  });
});
