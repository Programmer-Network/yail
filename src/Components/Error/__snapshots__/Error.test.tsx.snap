// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Error component > renders correctly - snapshot test 1`] = `
<DocumentFragment>
  <div
    class="yl-flex yl-min-h-full yl-flex-col"
    data-testid="error-wrapper"
  >
    <div
      class="yl-mx-auto yl-flex yl-w-full yl-flex-grow yl-flex-col yl-px-4 lg:yl-px-8"
    >
      <div
        class="yl-flex yl-flex-shrink-0 yl-justify-center"
      >
        <a
          class="yl-inline-flex"
        />
      </div>
      <div>
        <div
          class="yl-text-center"
        >
          <p
            class="yl-font-semibold yl-tracking-wide yl-text-primary md:yl-text-6xl"
            data-testid="error-text"
          >
            404 error
          </p>
          <h1
            class="yl-text-4xl yl-font-extrabold yl-tracking-tight yl-text-text sm:yl-text-3xl"
          >
            Page not found.
          </h1>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
